src/core/hash.zig

```diff
@@ -46,4 +49,13 @@ pub const Hash = struct {
         }
         return CompareResult.Equal;
     }
+
+    pub fn extend_and_hash(id: Hash, val: []u8, alloc: Allocator) !Self {
+        var hash_data = std.ArrayList(u8).init(alloc);
+        defer _ = hash_data.deinit();
```

Comment:
this should just be `defer hash_data.deinit()` - deinit is void

also can you init the array `withCapacity` since we know the length (then well only have one alloc) - also should use a fcn like insertSliceWithCapacity